python/grass/app/runtime.py

```diff
@@ -29,15 +29,34 @@ def get_grass_config_dir(major_version, minor_version, env):
     Determines path of GRASS GIS user configuration directory.
     """
     # The code is in sync with grass.app.runtime (but not the same).
+    env_dirname = "APPDATA" if WINDOWS else "HOME"
+    if env.get(env_dirname) is None:
+        raise KeyError(
```

Comment:
I can't quite see what exception would make sense here. `KeyError` is more related to the underlying dictionary rather than the state the error message is trying to describe. Consider [RuntimeError](https://docs.python.org/3/library/exceptions.html#RuntimeError).

Reply:
Make sense to me. Changed in 94a5369627.